libs/beacon/src/beacon_manager.cpp

```diff
@@ -119,20 +119,43 @@ std::pair<BeaconManager::Share, BeaconManager::Share> BeaconManager::GetReceived
   return shares_j;
 }
 
-void BeaconManager::AddCoefficients(MuddleAddress const &           from,
+bool BeaconManager::AddCoefficients(MuddleAddress const &           from,
                                     std::vector<Coefficient> const &coefficients)
 {
-  for (uint32_t ii = 0; ii <= polynomial_degree_; ++ii)
+  if (coefficients.size() == polynomial_degree_ + 1)
   {
-    C_ik[identity_to_index_[from]][ii].setStr(coefficients[ii]);
+    for (uint32_t ii = 0; ii <= polynomial_degree_; ++ii)
+    {
+      bool c_set{false};
+      C_ik[identity_to_index_[from]][ii].setStr(&c_set, coefficients[ii].data());
```

Comment:
Was the bool return value from this always an option in the library? Also, stylistically if there is no reason for ii I would just use i to maintain consistency

Reply:
The setStr here, why/how could it fail? I feel like it would be great if this fn could be reduced to:

```
if (coefficients.size() == polynomial_degree_ + 1)
{
  C_ik[identity[from]] = coefficients;
}
```

Or something like that

Reply:
Yes, the bool return value was always an option. The setStr can fail if someone sends you a string which does convert to number of the correct type i.e. in a  malformed message. There is no reason for ii, I can change it to i.